test/Microsoft.AspNet.Mvc.FunctionalTests/XmlDataContractSerializerInputFormatterTest.cs

```diff
@@ -0,0 +1,133 @@
+            var input = "<Store xmlns=\"http://schemas.datacontract.org/2004/07/XmlFormattersWebSite\"" +
+                        " xmlns:i=\"http://www.w3.org/2001/XMLSchema-instance\"><Address><State i:nil=\"true\"/>" +
+                        "<Zipcode>98052</Zipcode></Address><Customers><Customer><Address i:nil=\"true\"/><Id>1000" +
+                        "</Id></Customer></Customers><Id>10</Id></Store>";
+            var content = new StringContent(input, Encoding.UTF8, "application/xml-dcs");
+            var propertiesCollection = new List<KeyValuePair<string, string>>();
+            propertiesCollection.Add(new KeyValuePair<string, string>(nameof(Store.Id), typeof(Store).FullName));
+            propertiesCollection.Add(new KeyValuePair<string, string>(nameof(Address.Zipcode), typeof(Address).FullName));
+            propertiesCollection.Add(new KeyValuePair<string, string>(nameof(Customer.Id), typeof(Customer).FullName));
+            var expectedErrorMessages = propertiesCollection.Select(kvp =>
+            {
+                return string.Format(
+                    "{1}:Value type property '{0}' on type '{1}' has RequiredAttribute but " +
+                    "no DataMember(IsRequired = true) attribute.",
+                    kvp.Key, kvp.Value);
+            }).ToList();
+            expectedErrorMessages.Add("store.Address.State:The State field is required.");
+            expectedErrorMessages.Add("store.Customers[0].Address:The Address field is required.");
+
+            // Act
+            var response = await client.PostAsync("http://localhost/Validation/CreateStore", content);
+
+            //Assert
+            var dcsSerializer = new DataContractSerializer(typeof(ModelBindingInfo));
+            var responseStream = await response.Content.ReadAsStreamAsync();
+            var modelBindingInfo = dcsSerializer.ReadObject(responseStream) as ModelBindingInfo;
+            Assert.NotNull(modelBindingInfo);
+            Assert.NotNull(modelBindingInfo.Store);
+            Assert.Equal(10, modelBindingInfo.Store.Id);
+            Assert.NotNull(modelBindingInfo.Store.Address);
+            Assert.Equal(98052, modelBindingInfo.Store.Address.Zipcode);
+            Assert.Null(modelBindingInfo.Store.Address.State);
+            Assert.NotNull(modelBindingInfo.Store.Customers);
+            Assert.Equal(1, modelBindingInfo.Store.Customers.Count);
+            Assert.Equal(1000, modelBindingInfo.Store.Customers[0].Id);
+            Assert.Null(modelBindingInfo.Store.Customers[0].Address);
+            Assert.NotNull(modelBindingInfo.ModelStateErrorMessages);
+            Assert.Equal(expectedErrorMessages.Count(), modelBindingInfo.ModelStateErrorMessages.Count);
+            foreach (var expectedErrorMessage in expectedErrorMessages)
+            {
+                Assert.Contains(
+                modelBindingInfo.ModelStateErrorMessages,
```

Comment:
how do we know if this contains the exact same error message , and error message counts?